firmware/application/apps/ui_scanner.cpp

```diff
@@ -72,453 +74,251 @@ void ScannerThread::run() {
-					to_string_dec_uint(frequency_list[i])  );
-	desc_cycle.set( description_list[i] );
-}
-
-void ScannerView_AM::handle_retune(uint32_t i) {
-	text_cycle.set(	to_string_dec_uint(i + 1) + "/" +
-					to_string_dec_uint(frequency_list.size()) + ": " +
-					to_string_dec_uint(frequency_list[i])  );
-	desc_cycle.set( description_list[i] );
-}
+void ScannerView::handle_retune(uint32_t i) {
 
-void ScannerView_NFM::handle_retune(uint32_t i) {
 	text_cycle.set(	to_string_dec_uint(i + 1) + "/" +
-					to_string_dec_uint(frequency_list.size()) + ": " +
-					to_string_dec_uint(frequency_list[i]) );
+					to_string_dec_uint(frequency_list.size()) + ":" +
+					to_string_short_freq(frequency_list[i]) +"M"  );
 	desc_cycle.set( description_list[i] );
 }
 
-void ScannerView_AM::focus() {
-	field_lna.focus();
-}
-
-void ScannerView_FM::focus() {
-	field_lna.focus();
-}
-
-void ScannerView_NFM::focus() {
+void ScannerView::focus() {
 	field_lna.focus();
 }
 
-ScannerView_AM::~ScannerView_AM() {
-	audio::output::stop();
-	receiver_model.disable();
-	baseband::shutdown();
-}
-
-ScannerView_FM::~ScannerView_FM() {
-	audio::output::stop();
-	receiver_model.disable();
-	baseband::shutdown();
-}
-
-ScannerView_NFM::~ScannerView_NFM() {
+ScannerView::~ScannerView() {
 	audio::output::stop();
 	receiver_model.disable();
 	baseband::shutdown();
 }
 
-// NFM  same as before
+ScannerView::ScannerView(
+	NavigationView&,
+	const int32_t mod_type
+) : title_ { mod_name[mod_type] }
 
-ScannerView_NFM::ScannerView_NFM(
-	NavigationView&
-)
 {
+//	DEBUG(3, "ENTREE SCANNER: " + to_string_dec_uint(mod_type));
+	
 	add_children({
 		&rssi,
 		&labels,
 		&field_lna,
 		&field_vga,
 		&field_rf_amp,
 		&field_volume,
-		&field_bw,
 		&field_squelch,
 		&field_wait,
-		//&record_view,
 		&text_cycle,
 		&desc_cycle,
 		&text_cycle1,
 		&text_modulation,
-		//&waterfall,
 	});
 
-	size_t step_nfm = 0;
-	size_t	max_entry_nfm = 0 ;
-
-	std::string scanner_file = "SCANNER_NFM";
-	if (load_freqman_file(scanner_file, database)) {
-		for(auto& entry : database) {
-			// FIXME
-			if (entry.type == RANGE) {
+	size_t step = 0;
+	size_t	max_entry = 0 ;
+	size_t	def_step = 0 ;
+	size_t entry_full = 0 ;
+	std::string scanner_file ;
+	
+	switch (mod_type) {
+	case NFM: scanner_file = "SCANNER_NFM";	def_step = 12500;	add_children({&field_bw_NFM });				break;
+	case AM:  scanner_file = "SCANNER_AM";	def_step = 9000;	add_children({&field_bw_AM});				break;
+	case FM:  scanner_file = "SCANNER_FM";	def_step = 100000;	add_children({&field_bw_FM});				break;
+	case ANY: scanner_file = "SCANNER_ANY";	def_step = 10000;	add_children({&field_bw_ANY	});				break;
+	default: 																								break;
+}
+	// LEARN FREQUENCIES
+	
+	if ( load_freqman_file(scanner_file, database)  ) {
+		text_cycle1.set("SCANNER FILE PRESENT" );
+		for(auto& entry : database) {						// READ LINE PER LINE
+			
+			if ( (entry.type == RANGE)  && (entry_full == 0) )  {										//RANGE
```

Comment:
No need for all those parenthesis:

`if ( entry.type == RANGE && !entry_full ) {`

Will work fine